### ISSUE TITLE
whether you fixed the image size

### ISSUE THREAD

**Issue by chenjun2hao** [Jyouhou/ICDAR2019-ArT-Recognition-Alchemy#3]
Did you fix the size of the picture during the training? Or fill it with the widest possible image.

**Comment by PkuDavidGuan** [comment]
Yes, we fix the image size in our model training. We explore two schemes. 1) Resize all input images to 64*256. 2) Input images are padded into an n * n square, and n equals to the longer side of the original image.  The second scheme is slightly better than the first one.

**Comment by longshangbang** [comment]
This is an interesting issue. Let's keep it open.

@PkuDavidGuan has answered for me. I will give some more specification for the squarization.

The long sides are resized to 256 and short sides resized such that the aspect ratios remain unchanged. Then, we pad the short side with pure grey (127, 127, 127) to make a square image. 

You can refer to the paper that will be released soon.

**Comment by thomasryck** [comment]
Did you try other color than pure grey ?
I tried this techniques with another model architecture and I find out that the padding color had a significant impact on the model performance.

**Comment by longshangbang** [comment]
> Did you try other color than pure grey ?
> I tried this techniques with another model architecture and I find out that the padding color had a significant impact on the model performance.

How did you choose the "color"? Was the performance improved?

**Comment by thomasryck** [comment]
I tried just tried with 2 "colors": white and black but I could try more colors to have a real opinion on the impact of the color on the performance. I found out that using black was giving a more noisy output than using white for padding (e.g recognizing non existing character). By the way, in my case I am working with black text on black background mostly. So I guess that probably there is a link between the color you choose and the background of the text.

**Comment by longshangbang** [comment]
That's an interesting finding. 

What I have in mind is that, padding extra black may cause a shift in statistics for batch norm layer. Do you keep the padding the same for both training and evaluation?

**Comment by thomasryck** [comment]
In fact, I am using padding only for the evaluation and inference since my training are synth images and real images of same size. I forgot to say that my model is an end-to-end detection and recognition model. And that probably the most important influence is not from the recognition model but I think that in any case the color padding as an influence on recognition network performance and should be investigated. For now, I have no time to deep dive in this kind of experimentation. And maybe it could be interesting to use the same padding as U-Net by mirroring the missing parts.

**Comment by thomasryck** [comment]
Probably the quantity of padding could have an impact too. Number of pixels corresponding to padding

**Comment by longshangbang** [comment]
Different setting for training and testing may not be a good idea, though. 

I am curious why you need padding in evaluation?

**Comment by thomasryck** [comment]
Yes I know 😓 

In fact, I have only one sort of labeled training data but I have two possible use case and so kind of input images. And on the case not available for training, the image sizes are not compatible with my network fixed sized. That's why I need to use padding for that use case.

**Comment by longshangbang** [comment]
I see. That's still a good attempt. Or maybe you can try to concatenate a natural image?
